type_infer/infer.py

```diff
@@ -128,17 +128,65 @@ def type_check_sequence(element: object) -> str:
 
 
 def type_check_date(element: object) -> str:
-    try:
-        dt = pd.to_datetime(element)
-
-        # Not accurate 100% for a single datetime str, but should work in aggregate
-        if dt.hour == 0 and dt.minute == 0 and dt.second == 0 and len(str(element)) <= 16:
-            return dtype.date
+    """
+    Check if element corresponds to a date-like object.
+    """
+    # check if element represents a date (no hour/minute/seconds)
+    is_date = False
+    # check if element represents a datetime (has hour/minute/seconds)
+    is_datetime = False
+    # check if it makes sense to convert element to unix time-stamp by
+    # evaluating if, when converted, the element represents a number that
+    # is compatible with a Unix timestamp (number of seconds since 1970-01-01T:00:00:00)
+    # note that we also check the number is not larger than the "epochalypse time",
+    # which is when the unix timestamp becomes larger than 2^32 - 1 seconds. We do
+    # this because timestamps outside this range are likely to be unreliable and hence
+    # rather treated as every-day numbers.
+    min_dt = pd.to_datetime('1970-01-01 00:00:00', utc=True)
+    max_dt = pd.to_datetime('2038-01-19 03:14:08', utc=True)
+    valid_units = ['ns', 'us', 'ms', 's', 'D']
+    for unit in valid_units:
+        # Yes, some people still use Julian Days...
+        if unit == 'D':
```

Comment:
Code here looks slightly duplicated. I suggest setting `origin` to either `julian` or `unix` within the if-else, then a common try-except clause for all units.

Reply:
That's correct. I can fix this by iterating over a dictionary (unit - origin) instead of just a list of units to plainly avoid the if-else clause. I'll implement this change and push it asap. Nice catch!